sdk/core/Azure.Core/tests/TestFramework/MockRequest.cs

```diff
@@ -17,6 +17,9 @@ public MockRequest()
 
         private readonly Dictionary<string, List<string>> _headers = new Dictionary<string, List<string>>(StringComparer.OrdinalIgnoreCase);
 
+#if HAS_INTERNALS_VISIBLE_CORE
+internal
```

Comment:
Why not have these always be internal?

Reply:
Nevermind I think I understand as these are overrides of virtual methods. 

Of course I would argue that the test projects should not have internals visible to but that is an argument for another time.

Reply:
I'm also curious why these https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Request.cs#L25 is `protected internal`, do we really need to call them from other places only inside Azure.Core?

Reply:
Yes, https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/RequestHeaders.cs wraps the `Request` to provide more user-friendly API.